apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/lib/site-intent/use-site-intent.js

```diff
@@ -2,17 +2,24 @@ import apiFetch from '@wordpress/api-fetch';
 import { useState, useEffect, useCallback } from '@wordpress/element';
 
 const useSiteIntent = () => {
-	const [ siteIntent, setSiteIntent ] = useState( '' );
+	const [ siteIntent, setSiteIntent ] = useState( undefined );
```

Comment:
For those who have not set site intent, `/wpcom/v2/site-intent` returns an empty string, so I set the initial state to `undefined` to distinguish it from the initial state, failure, etc.
(If we set an empty string as the initial value, we don't have ways to know if the user has not set the site intent or if the fetch just failed.)